### ISSUE TITLE
`prefer-spread` flags non-array concat

### ISSUE THREAD

**Issue by kasugaiasuka** [sindresorhus/eslint-plugin-unicorn#1070]
`prefer-spread` affects not only `Array#concat()` but also `AnyClassThatIsNotArray#concat()`.

Some array utilities will be false positives according to this rule.
(for example, lazy.js, linq.js, and so on.)

`prefer-spread`

```js
class Foo {
	concat(array) {
		doSomething(array);
	}
}

// this line is flagged even if this could not be written using spread operator
new Foo().concat([1, 2, 3]);

// if autofixed, the line become like this
// these are not semantically equal
[...new Foo(), 1, 2, 3];
```

**Comment by fisker** [comment]
Duplicated of #1064
